src/services/image.service.ts

```diff
@@ -0,0 +1,34 @@
+import type { Playlist, PlaylistItem } from '#types/playlist';
+import { findPlaylistImageForWidth } from '#src/utils/collection';
+import type { Config } from '#types/Config';
+
+export type ImageProperty = 'shelfImage' | 'backgroundImage' | 'channelLogoImage';
+
+export const generateAlternateImageURL = (mediaid: string, imageLabel: string, width: number) =>
+  `https://img.jwplayer.com/v1/media/${mediaid}/images/${imageLabel}.jpg?width=${width}`;
+
+export const getAlternateImageLabel = (config: Config, propertyName: ImageProperty, item: PlaylistItem, playlist?: Playlist) => {
+  const alternateImageLabel = propertyName === 'shelfImage' ? playlist?.[propertyName] : item[propertyName];
```

Comment:
Nice to have a small comment to explain the difference here
```suggestion
  // Shelf images are set at the playlist level, all others are at the media item level
  const alternateImageLabel = propertyName === 'shelfImage' ? playlist?.[propertyName] : item[propertyName];
```